lib/src/coap_media_type.dart

```diff
@@ -149,7 +221,11 @@ class CoapMediaType {
   /// Gets the file extension of the given media type.
   static String fileExtension(int mediaType) {
     if (_registry.containsKey(mediaType)) {
-      return _registry[mediaType]![1];
+      if (_registry[mediaType]!.length > 1) {
+        return _registry[mediaType]![1];
+      } else {
+        return 'undefined';
```

Comment:
I had a another look at the IANA registry but there does not seem to be an explicit "undefined" extension. Would returning `null` be an option for future versions? For now the API change is probably not desirable.

Reply:
Now we are in the world of null safety I would prefer not to return a null, I've got into the habit now of returning an empty string if no other value can be calculated. You could just annotate the API comment to say that 'undefined' can be returned if the media type cannot be resolved, as long as we inform people what we are doing I think its OK.